ls1intum/Artemis
# Chore/remove transactional

This PR removes the annotation `@Transactional` where possible. The issue with `@Transactional` is that it might lead to unwanted save operations in the database which can lead to very weird bugs that are hard to track.

**Important:** `@Transactional` should only be used for more complex `delete` and `create` operations where it makes sense.

### Checklist

### Steps for Testing
All views and actions in the client should be quickly tested, in particular if they are related to:
1. Functionality related to users (UserService) on both configurations: Jira, Bitbucket, Bamboo on `TS1` and Gitlab, Jenkins on `TS2`
    * registration
    * login
    * logout
    * user administration (create, edit, delete, change groups, change authorities)
    * participate in guided tours (and check if the last step is saved correctly)
    * course registration 
    * course groups changes (add/remove students, tutors, instructors)
    * create courses
2. Complaints



### Test Coverage
no changes

### Screenshots
no UI changes
